gci/linter_repos.py

```diff
@@ -0,0 +1,93 @@
+import re
+from multiprocessing import Pool
+
+from IGitt.GitHub.GitHub import GitHub, GitHubToken
+
+re_gh = r'https?://github\.com'
+re_gh_repo = r'https?://github\.com/([\w-]+/[\w-]+)'
+
+
+def get_github(token):
+    return GitHub(GitHubToken(token))
+
+
+def get_repo_host(url):
+    if re.match(re_gh, url):
+        return 'github'
+
+
+def get_repo_name(url, host):
+    if host == 'github':
+        result = re.match(re_gh_repo, url)
+        return result and result.groups()[0]
+
+
+def simplify_issue(issue, repo_url=''):
+    return {
+      'id': issue.number,
+      'repo_url': repo_url,
+      'title': issue.title,
+      'labels': list(issue.labels),
+      'description': issue.description,
+    }
+
+
+def _lower_set(iterable):
+    return set(map(str.lower, iterable))
+
+
+def issue_filter(issue, labels, desc_keywords):
+    norm_issue_labels = _lower_set(issue['labels'])
+    norm_labels = _lower_set(labels)
+
+    if len(norm_issue_labels & norm_labels) > 0:
+        return True
+
+    norm_keywords = _lower_set(desc_keywords)
+    norm_text = str.lower(issue['title'] + ' ' + issue['description'])
+    norm_text = set(norm_text.split())
+
+    if len(norm_keywords & norm_text) > 0:
+        return True
+
+
+def fetch_issues(hosts, targets, labels=[],
+                 desc_keywords=[], threads=8, debug=False):
+
+    all_repo_issues = {}
+    github = hosts.get('github')
+
+    for target in targets:
+        if debug:
+            print('processing', target)
+
+        try:
+            host = get_repo_host(target)
+
+            if host == 'github':
+                gh_repo_name = get_repo_name(target, host)
+                gh_repo = github.get_repo(gh_repo_name)
```

Comment:
use `community.git.get_irepo` , and `giturlparse.parse(url)` , and add any other git / hoster utility functions in there.